api/client.go

```diff
@@ -670,24 +709,41 @@ func (c *Client) SetBackoff(backoff retryablehttp.Backoff) {
 // Also, only the client's config is currently copied; this means items not in
 // the api.Config struct, such as policy override and wrapping function
 // behavior, must currently then be set as desired on the new client.
+//
+// One exception to this is the addr field of Client, which is also copied over.
+// Otherwise you have the confusing result of creating a client, customizing
+// the address via client.SetAddress() then cloning it and having it show the
+// default address again, instead of the one you just set. I'm not even sure
```

Comment:
I don't think this comment is the right place to pose such questions ("I'm not even sure...").  My guess is that yes, it's a convenience/optimization.

Reply:
I'm not sure this comment update even needs to be added.  The address is part of the config, so the existing comment implies it should be in scope of Clone.  The fact that there's some additional work we need to do such that SetAddress is not necessary post-Clone isn't really relevant to the consumer of the doc for this method.  If you want to document why we do SetAddress, I'd put that as an inline comment above where we invoke it.

Reply:
Sounds good